### PR TITLE
Gse install

### PR DESCRIPTION
**Pull request by CBiasuzzi**
distribution module + install.sh (default is on top of an existing powsybl installation) + gse script (to run gse-demo)

### CODE REVIEW

File: install.sh [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+    echo "  install                  Compile GSE modules and install it (default target)"
+    echo "  help                     Display this help"
+    echo "  docs                     Generate the documentation (Doxygen/Javadoc)"
+    echo ""
+    echo "GSE options:"
+    echo "  --help                   Display this help"
+    echo "  --prefix                 Set the installation directory (default is $HOME/powsybl)"
+    echo "  --package-type           Set the package format. The supported formats are zip, tar, tar.gz and tar.bz2 (default is zip)"
+    echo "  --skip-tests             compile modules without testing"
+    echo "  --with-tests             compile modules with testing (default)"
```
Comment (mathbagu) [inline install.sh]: Is it necessary to add this option: the default value does the job

File: gse-scripts/gse [hunk 2]
```diff
@@ -0,0 +1,50 @@
+#!/bin/bash
```
Comment (mathbagu) [inline gse-scripts/gse]: You have to add a gse.bat for windows

File: gse-distribution/src/main/resources/assemblies/full.xml [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+            <outputDirectory>doc/java/ipst-core</outputDirectory>
+            <fileMode>0644</fileMode>
+        </fileSet>
+
+        <!-- Doxygen -->
+        <fileSet>
+            <directory>${project.basedir}/../build/html</directory>
+            <outputDirectory>doc/cpp/ipst-core</outputDirectory>
+            <fileMode>0644</fileMode>
+        </fileSet>
```
Comment (mathbagu) [inline gse-distribution/src/main/resources/assemblies/full.xml]: There is no doxygen in GSE

File: gse-distribution/src/main/resources/assemblies/full.xml [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+            <includes>
+                <include>*.so</include>
+            </includes>
+        </fileSet>
+
+
+        <!-- Javadoc -->
+        <fileSet>
+            <directory>${project.basedir}/../target/site/apidocs</directory>
+            <outputDirectory>doc/java/ipst-core</outputDirectory>
```
Comment (mathbagu) [inline gse-distribution/src/main/resources/assemblies/full.xml]: Fix this outputDirectory to doc/java/powsybl-gse

File: gse-distribution/src/main/resources/assemblies/full.xml [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+    <includeBaseDirectory>true</includeBaseDirectory>
+    <dependencySets>
+        <dependencySet>
+            <outputDirectory>share/java</outputDirectory>
+        </dependencySet>
+    </dependencySets>
+    <!-- include scripts, log configurations and binaries, if available -->
+    <fileSets>
+        <fileSet>
+            <directory>${project.basedir}/../gse-scripts</directory>
```
Comment (mathbagu) [inline gse-distribution/src/main/resources/assemblies/full.xml]: Rename the gse-scripts directory to scripts

File: gse-distribution/src/main/resources/assemblies/full.xml [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+            <outputDirectory>bin</outputDirectory>
+            <includes>
+                <include>*</include>
+            </includes>
+            <fileMode>0755</fileMode>
+        </fileSet>
+        <fileSet>
+            <directory>${project.basedir}/../gse-config</directory>
```
Comment (mathbagu) [inline gse-distribution/src/main/resources/assemblies/full.xml]: Remove the gse-config directory to config

File: gse-distribution/src/main/resources/assemblies/full.xml [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+            <outputDirectory>etc</outputDirectory>
+            <includes>
+                <include>logback*.xml</include>
+            </includes>
+            <fileMode>0644</fileMode>
+        </fileSet>
+        <fileSet>
```
Comment (mathbagu) [inline gse-distribution/src/main/resources/assemblies/full.xml]: Remove this section: there is no C++ binary in GSE

File: gse-distribution/src/main/resources/assemblies/full.xml [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+            <directory>${project.basedir}/../build/bin</directory>
+            <outputDirectory>bin</outputDirectory>
+            <includes>
+                <include>*</include>
+            </includes>
+            <fileMode>0755</fileMode>
+        </fileSet>
+        <fileSet>
```
Comment (mathbagu) [inline gse-distribution/src/main/resources/assemblies/full.xml]: Remove this section: there is no C++ libraries in GSE

**Comment by geofjamg** [comment]
GSE is a UI framework and has not to be "itools" packaged. JavaFX packager is a most common way to package a JavaFX application.

**Pull request closed without merging**